utils/coreutils/techutils.go

```diff
@@ -88,51 +105,51 @@ var technologiesData = map[Technology]TechData{
 		applicabilityScannable:     true,
 	},
 	Pnpm: {
-		indicators:                 []string{"pnpm-lock.yaml"},
+		indicators:                 map[string]ContentValidator{"pnpm-lock.yaml": nil},
 		exclude:                    []string{".yarnrc.yml", "yarn.lock", ".yarn"},
 		packageDescriptors:         []string{"package.json"},
 		packageVersionOperator:     "@",
 		packageInstallationCommand: "update",
 		applicabilityScannable:     true,
 	},
 	Yarn: {
-		indicators:             []string{".yarnrc.yml", "yarn.lock", ".yarn", ".yarnrc"},
+		indicators:             map[string]ContentValidator{".yarnrc.yml": nil, "yarn.lock": nil, ".yarn": nil, ".yarnrc": nil},
 		exclude:                []string{"pnpm-lock.yaml"},
 		packageDescriptors:     []string{"package.json"},
 		packageVersionOperator: "@",
 		applicabilityScannable: true,
 	},
 	Go: {
-		indicators:                 []string{"go.mod"},
+		indicators:                 map[string]ContentValidator{"go.mod": nil},
 		packageDescriptors:         []string{"go.mod"},
 		packageVersionOperator:     "@v",
 		packageInstallationCommand: "get",
 	},
 	Pip: {
 		packageType:            Pypi,
-		indicators:             []string{"setup.py", "requirements.txt"},
-		packageDescriptors:     []string{"setup.py", "requirements.txt"},
-		exclude:                []string{"Pipfile", "Pipfile.lock", "pyproject.toml", "poetry.lock"},
+		indicators:             map[string]ContentValidator{"pyproject.toml": pyProjectTomlIndicatorContent(Pip), "setup.py": nil, "requirements.txt": nil},
```

Comment:
I think the current implementation is good - but IMO adding a key that'll indicate it - will be a bit more cleaner than having keys referring to nil values, most of the team.
The new key as it could be empty other than python packages - and only if relevant it'll be checked.
wdyt?